### PR TITLE
Deprecate featureflags.FlagValue.SetSlice, unnecessary public

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: service/featuregate/flags_test.go [hunk 1]
```diff
@@ -96,33 +109,3 @@ func TestFlagValue_withPlus(t *testing.T) {
...
-		},
-		{
-			name:     "repeated items",
-			input:    []string{"foo", "-bar", "-foo"},
-			expected: FlagValue{"foo": true, "bar": false},
-		},
-	} {
-		t.Run(tc.name, func(t *testing.T) {
-			v := FlagValue{}
-			assert.NoError(t, v.SetSlice(tc.input))
```
Comment (jpkrohling) [inline service/featuregate/flags_test.go:124]: Could you please keep this, removing only when the public function is removed?

Comment (bogdandrutu) [reply service/featuregate/flags_test.go:124]: Why? We test that via the "comma separated" cases, so the only untested code is the call from `SetSlice` to `setSlice`.

Comment (jpkrohling) [reply service/featuregate/flags_test.go:124]: Which is exactly what is being deprecated :-)

Comment (bogdandrutu) [reply service/featuregate/flags_test.go:124]: I don't think I understand:

```golang
// fully tested
func Set() {
  // calls setSlice()
}

func SetSlice() {
  // Not tested 1 line
  return setSlice()
}

// fully tested
func setSlice() {}
}
```

Comment (bogdandrutu) [reply service/featuregate/flags_test.go:124]: The "setSlice" is very well tested, it is only that deprecated func that is not tested which does very little (to nothing).

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**